hedera-node/hedera-app/src/test/java/com/hedera/node/app/grpc/impl/usage/GrpcUsageTrackerTest.java

```diff
@@ -0,0 +1,226 @@
+        final ConfigProvider configProvider = mock(ConfigProvider.class);
+        final VersionedConfiguration configuration = mock(VersionedConfiguration.class);
+        final ServerCall<String, String> serverCall = mock(ServerCall.class);
+        final ServerCallHandler<String, String> handler = mock(ServerCallHandler.class);
+        final Metadata metadata = new Metadata();
+        metadata.put(userAgentHeaderKey, "hiero-sdk-java/2.3.1");
+        final MethodDescriptor<String, String> descriptor = newDescriptor("proto.MyService/commit");
+
+        when(configProvider.getConfiguration()).thenReturn(configuration);
+        when(configuration.getConfigData(GrpcUsageTrackerConfig.class)).thenReturn(config);
+        when(serverCall.getMethodDescriptor()).thenReturn(descriptor);
+
+        final GrpcUsageTracker usageTracker = new GrpcUsageTracker(configProvider, clock);
+
+        assertDoesNotThrow(() -> usageTracker.interceptCall(serverCall, metadata, handler));
+
+        // get the usage bucket... there should be no data captured in it since usage tracking is disabled
+        final AtomicReference<UsageBucket> bucketRef =
+                (AtomicReference<UsageBucket>) usageBucketRefHandle.get(usageTracker);
+        assertNotNull(bucketRef);
+        final UsageBucket usageBucket = bucketRef.get();
+        assertNotNull(usageBucket);
+        assertTrue(usageBucket.usageData().isEmpty());
+        assertEquals(usageBucket.time(), Instant.parse("2025-04-03T15:30:00.000Z"));
+    }
+
+    @Test
+    void testInterceptEnabled() {
+        final Clock clock = Clock.fixed(Instant.parse("2025-04-03T15:32:32.426457Z"), ZoneOffset.UTC);
+        final GrpcUsageTrackerConfig config = new GrpcUsageTrackerConfig(true, 15, 100);
+        final ConfigProvider configProvider = mock(ConfigProvider.class);
+        final VersionedConfiguration configuration = mock(VersionedConfiguration.class);
+        final ServerCall<String, String> serverCall = mock(ServerCall.class);
+        final ServerCallHandler<String, String> handler = mock(ServerCallHandler.class);
+        final Metadata metadata = new Metadata();
+        metadata.put(userAgentHeaderKey, "hiero-sdk-java/2.3.1");
+        final MethodDescriptor<String, String> descriptor = newDescriptor("proto.MyService/commit");
+
+        when(configProvider.getConfiguration()).thenReturn(configuration);
+        when(configuration.getConfigData(GrpcUsageTrackerConfig.class)).thenReturn(config);
+        when(serverCall.getMethodDescriptor()).thenReturn(descriptor);
+
+        final GrpcUsageTracker usageTracker = new GrpcUsageTracker(configProvider, clock);
+
+        // make two calls
+        assertDoesNotThrow(() -> usageTracker.interceptCall(serverCall, metadata, handler));
+        assertDoesNotThrow(() -> usageTracker.interceptCall(serverCall, metadata, handler));
+
+        // get the usage bucket... there should be no data captured in it since usage tracking is disabled
```

Comment:
This comment looks like a copy-paste error since the interceptor is enabled for this test